model/activist.go

```diff
@@ -237,12 +237,47 @@ func GetOrCreateUser(db *sqlx.DB, name string) (User, error) {
 	}
 
 	// There was an error, so try inserting the user first.
-	_, err = db.Exec("INSERT INTO activists (name) VALUES (?)", name)
+	// Wrap in transaction to avoid issue where a new user
+	// is inserted successfully, but we are unable to retrieve
+	// the new user, which will leave database in inconsistent state
+
+	tx, err := db.Beginx()
+	if err != nil {
+		return User{}, errors.Wrap(err, "Failed to create transaction")
+	}
+
+	_, err = tx.Exec("INSERT INTO activists (name) VALUES (?)", name)
 	if err != nil {
+		tx.Rollback()
 		return User{}, errors.Wrapf(err, "failed to insert user %s", name)
 	}
 
-	return GetUser(db, name)
+	query := `
+SELECT
```

Comment:
Can `getUsers` and this use the same "SELECT ... FROM ACTIVISTS" part of the query?